libraries/support/azure/rest.rb

```diff
@@ -7,18 +7,20 @@ module Azure
   class Rest
     attr_reader :host, :resource, :credentials
 
-    def initialize(host, credentials: {})
-      @host        = host
-      @resource    = "#{host}/"
-      @credentials = credentials
+    def initialize(client)
+      @host        = client.base_url
+      @resource    = client.base_url.to_s
+      @resource    += '/' unless @resource[-1, 1] == '/'
```

Comment:
What's the intent of this line?

Reply:
Exactly - thanks for the cleaner version @TrevorBramble